includes/payment-methods/class-wc-stripe-express-checkout-helper.php

```diff
@@ -543,6 +543,14 @@ public function should_show_express_checkout_button() {
 			return false;
 		}
 
+		// Don't show if the total price is 0.
+		// ToDo: support free trials. Free trials should be supported if the product does not require shipping.
+		if ( 0.0 === (float) WC()->cart->get_total( false )
+			|| ( $this->is_product() && 0.0 === (float) $this->get_product()->get_price() )
```

Comment:
When my cart total is 0 (i.e. empty cart, free product, or with big enough coupon), and I visit the product page for a product that is not free, I think we still want to have express payment available. Same with Pay for Order page.

Something like?

```suggestion
		if ( ( ( is_cart() || is_checkout() ) && 0.0 === (float) WC()->cart->get_total( false ) )
			|| ( $this->is_product() && 0.0 === (float) $this->get_product()->get_price() )
```

Reply:
That's a good catch 👍